lib/pymedphys/_experimental/wlutz/findbb.py

```diff
@@ -43,41 +59,76 @@ def find_bb_centre(
 
 
 def optimise_bb_centre(
-    field: imginterp.Field, bb_diameter, field_centre, initial_bb_centre=None
+    field: imginterp.Field, bb_diameter, field_centre, initial_bb_centre=None, repeats=2
 ):
     if initial_bb_centre is None:
         initial_bb_centre = field_centre
 
     search_square_edge_length = bb_diameter / np.sqrt(2) * 0.8
-    bb_centre = _minimise_bb(
-        field, bb_diameter, search_square_edge_length, initial_bb_centre
+    all_centre_predictions = np.array(
+        _bb_finding_repetitions(
+            field, bb_diameter, search_square_edge_length, initial_bb_centre
+        )
     )
+    median_of_predictions = np.nanmedian(all_centre_predictions, axis=0)
+
+    diff = np.abs(all_centre_predictions - median_of_predictions)
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore", category=RuntimeWarning)
+        within_tolerance = np.all(diff < BB_REPEAT_TOL, axis=1)
 
-    all_centre_predictions = [bb_centre]
-    for bb_size_factor in [0.5, 0.6, 0.7, 0.8, 0.9]:
+    assert len(within_tolerance) == len(BB_SIZE_FACTORS_TO_SEARCH_OVER)
+
+    if np.all(within_tolerance):
+        return median_of_predictions
+
+    if repeats == 0:
+        raise ValueError("Unable to determine BB position within designated repeats")
+
+    out_of_tolerance = np.invert(within_tolerance)
+    if np.sum(out_of_tolerance) > len(BB_SIZE_FACTORS_TO_SEARCH_OVER) / 3:
```

Comment:
Is this divide by three to address x, y, z or something else?

Reply:
It's a little bit of a "magic number", essentially the ball bearing is found independently 11 times each time with a slightly different input parameter. If more than a third of them deviate from the median by more than 0.2 mm then raise an error.

Here are the 11 different parameters used:

https://github.com/pymedphys/pymedphys/blob/56dcdf8b29f7d7773551aa343f70fb3e5c45275a/lib/pymedphys/_experimental/wlutz/findbb.py#L26-L38

Reply:
In order to finally exit the recursion though every variation all needs to agree with the median within the tolerance of 0.2 mm:

https://github.com/pymedphys/pymedphys/blob/56dcdf8b29f7d7773551aa343f70fb3e5c45275a/lib/pymedphys/_experimental/wlutz/findbb.py#L82-L83